src/inmanta/module.py

```diff
@@ -346,6 +345,14 @@ def to_list(cls, v: object) -> object:
     @validator("requires", pre=True)
     @classmethod
     def requires_to_list(cls, v: object) -> object:
+        if isinstance(v, dict):
+            # transform legacy format for backwards compatibility
+            result: List[str] = []
+            for key, value in v.items():
+                if not (isinstance(key, str) and isinstance(value, str) and value.startswith(key)):
+                    raise ValueError("Invalid legacy requires format, expected `mod: mod [constraint]`.")
+                result.append(value)
```

Comment:
The old implementation contained a bug. I know for sure that the version constraint was not taken into account if you would use the _"legacy format"_. I had a look as the source code and if I am not mistaken, only the name of the module was taken into account. Do you agree?
```suggestion
                result.append(key)
```

Reply:
I did a test and it seems that the requirement is ignored entirely

Reply:
I'll have a look now, but even if that is the case the question is whether we want to stay compatible with the old behavior or just the old format (fixing the bug). You think we should model the same behavior?
@bartv any opinion on this?

Reply:
(I've looked at the implementation and I agree with your assessment that there was a bug there)

Reply:
(I've confirmed it with a test case.)

Reply:
I have no strong opinion on whether we should maintain compatible with the old behavior or the old form. Currently we have some unverified version constraint in the module sets. We should at least verify that the module sets don't contains a conflict when we would patch the requirements validation in this PR.